galoy-client/src/client/mod.rs

```diff
@@ -328,27 +308,44 @@ impl GaloyClient {
             target_confirmations: Some(target_conf),
             wallet_id: self.btc_wallet_id.clone(),
         };
-        let response =
-            post_graphql::<StablesatsOnChainTxFee, _>(&self.client, &self.config.api, variables)
-                .await?;
+        let response = GaloyClient::traced_gql_request::<StablesatsOnChainTxFee, _>(
+            &self.client,
+            &self.config.api,
+            variables,
+        )
+        .await?;
         if response.errors.is_some() {
             if let Some(error) = response.errors {
                 return Err(GaloyClientError::GraphQLApi(error[0].clone().message));
             }
         }
 
         let response_data = response.data;
-        let result = match response_data {
-            Some(data) => data,
-            None => {
-                return Err(GaloyClientError::GraphQLApi(
-                    "Empty `onChainTxFee` in response data".to_string(),
-                ))
-            }
-        };
+        let result = response_data.ok_or_else(|| {
+            GaloyClientError::GraphQLApi("Empty `onChainTxFee` in response data".to_string())
+        })?;
 
         let onchain_tx_fee = StablesatsTxFee::try_from(result)?;
 
         Ok(onchain_tx_fee)
     }
+
+    async fn traced_gql_request<Q: GraphQLQuery, U: reqwest::IntoUrl>(
```

Comment:
if you make this an instance method you could avoid the duplication of passing in `self.client` and `self.config.api` everywhere - not a big deal.